lib/taskana-core/src/test/java/pro/taskana/database/TestDataGenerator.java

```diff
@@ -51,12 +55,21 @@ public void generateTestData(DataSource dataSource) throws SQLException {
             runner.setStopOnError(true);
             runner.setLogWriter(this.logWriter);
             runner.setErrorLogWriter(this.errorLogWriter);
-            runner.runScript(
-                new InputStreamReader(this.getClass().getResourceAsStream("/sql/classification.sql")));
-            runner.runScript(new InputStreamReader(this.getClass().getResourceAsStream("/sql/workbasket.sql")));
-            runner.runScript(new InputStreamReader(this.getClass().getResourceAsStream("/sql/task.sql")));
-            runner.runScript(
-                new InputStreamReader(this.getClass().getResourceAsStream("/sql/workbasket-access-list.sql")));
+            if (isDb2) {

```

Comment:
I'd prefer to replace false/true to 0/1 in memory instead of creating extra sql files for that. This way we don't have to worry about inconsistency about those files. generateMonitoringSqlData() is a good example. 
A step further would be to save those Strings in static variables so that the replacement is only necessary once. This assumes that the datasource does not change in one run (which it does not)

Reply:
I agree. Don't you have it available already?

Reply:
Yes. I have already implemented it. I'll add a commit to this PR in a little.

Reply:
Ok, so I don't do anything more with this request. I guess, you will also add the static method TaskanaEngineImpl.isDb2(). If you are explicitely asking whether it's DB2 you don't need a parameter on this method ...

Reply:
By the time when you call the isDb2() method, TaskanaEngineImpl has already been initialized and especially, createSqlSessionManager() has already been executed. This method determines the database system and could save either the db system name or a 'isDb2' flag statically.
But it is prudent to not assume this order of execution and pass a connection to the isDB2() method. In this way you are on the safe side ...